Auto Layout/CollectionViewController.swift

```diff
@@ -57,25 +87,47 @@ final private class CollectionViewDataSource: NSObject, UICollectionViewDataSour
     self.cellCount = cellCount
   }
 
-  func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+  func numberOfSections(in collectionView: UICollectionView) -> Int {
     return cellCount
   }
 
+  func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+    if section.isCarouselSection {
+      return 10
+    }
+    return 1
+  }
+
   func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
     let headline = ContentGenerator.words(min: 2,8)
     let summary = ContentGenerator.words(min: 20,40)
 
-    switch indexPath.row % 2 {
-    case 0:
+    if indexPath.section.isCarouselSection {
+      let cell: LargeImageCell = collectionView.dequeue(for: indexPath)
+      cell.updateUI(
+        headline: "Miles Davis",
+        summary: "Miles Dewey Davis III was an American jazz trumpeter, bandleader, and composer.",
+        image: "miles.png"
+      )
+      return cell
+    } else if indexPath.section.isHeadlineSummarySection {
       let cell: HeadlineSummaryCell = collectionView.dequeue(for: indexPath)
       cell.set(headline: headline, summary: summary)
       return cell
-    case 1:
+    } else {
       let cell: ThumbnailCell = collectionView.dequeue(for: indexPath)
       cell.set(headline: headline, summary: summary)
       return cell
-    default:
-      fatalError()
     }
   }
 }
+
+extension Int {
+  var isCarouselSection: Bool {
+    return (self + 1) % 5 == 0
+  }
+
+  var isHeadlineSummarySection: Bool {
+    return (self + 1) % 4 == 0
+  }
+}
```

Comment:
should we add `isThumbnailSection` too and use it? then we can catch logical errors in the above `else` and `fatalError` instead of defaulting to the thumbnailSection.. not that it matters too much here hah